lib/VM/JSLib/Intl.cpp

```diff
@@ -1395,15 +1425,27 @@ void toDateTimeOptions(platform_intl::Options &options, int dtoFlags) {
       }
     }
   }
-  if (!needDefaults) {
-    return;
-  }
-  for (const OptionData &pod : kDTFOptions) {
-    if ((dtoFlags & kDTODate && pod.flags & kDateDefault) ||
-        (dtoFlags & kDTOTime && pod.flags & kTimeDefault)) {
-      options.emplace(pod.name, std::u16string(u"numeric"));
+
+  const bool hasStyle = options.count(u"dateStyle") > 0 || options.count(u"timeStyle") > 0;
+  if (hasStyle)
+    needDefaults = false;
+
+  if (!(dtoFlags & kDTOTime) && options.count(u"timeStyle") > 0)
+    return ExecutionStatus::EXCEPTION;
```

Comment:
this is not how we use ```ExecutionStatus```. You should modify this function to take a Runtime, and then use ```Runtime::raiseTypeError``` here.

Reply:
Absolutely ..